R/SOS_200-methods-parsing.R

```diff
@@ -27,46 +27,318 @@
+  .intendedApplication <- sapply(.sosOffering[sosIntendedApplicationName], xmlValue)
+  if(length(.intendedApplication) == 0) .intendedApplication <- NA_character_
+
+  .env <- .sosOffering[[sosObservedAreaName]][[gmlEnvelopeName]]
+  if(!is.null(.env)) {
+    .observedArea <- list(
+      srsName = xmlGetAttr(.env, "srsName"),
+      lowerCorner = xmlValue(.env[[gmlLowerCornerName]]),
+      upperCorner = xmlValue(.env[[gmlUpperCornerName]]))
+
+    if(sosSwitchCoordinates(sos)) {
+      warning("Switching coordinates in envelope of ObservationOffering!")
+      .origLC <- strsplit(x = .observedArea[["lowerCorner"]], split = " ")
+      .lC <- paste(.origLC[[1]][[2]], .origLC[[1]][[1]])
+      .origUC <- strsplit(x = .observedArea[["upperCorner"]], split = " ")
+      .uC <- paste(.origUC[[1]][[2]], .origUC[[1]][[1]])
+      .observedArea <- list(srsName = xmlGetAttr(.env, "srsName"),
+                            lowerCorner = .lC, upperCorner = .uC)
+    }
+
+    if(sos@verboseOutput)
+      cat("[parseSosObservationOffering_200] boundedBy:",
+          toString(.observedArea), "\n")
+  }
+  else {
+    .observedArea <- list()
+  }
+
+  # warn if time or envelope is missing -> probably sensor without data.
+  .warningText <- ""
+  if(length(.observedArea) < 1) {
+    .warningText <- "\t'sos:observedArea' is NA/empty.\n"
+  }
+  if(extends(class(.resultTime), "GmlTimeInstant") &&
+     is.na(.resultTime@timePosition@time)) {
+    .warningText <- paste(.warningText, "\t'sos:resultTime' is NA/empty.\n")
+  }
+  if(extends(class(.phenomenonTime), "GmlTimeInstant") &&
+     is.na(.phenomenonTime@timePosition@time)) {
+    .warningText <- paste(.warningText, "\t'sos:phenomenonTime' is NA/empty.\n")
+  }
+  if(length(.warningText) > 1) {
+    warning(paste("Error when parsing offering '", .id, "':\n",
+                  .warningText, sep = ""))
+  }
+
+  .ob <- new("SosObservationOffering_2.0.0", id = .id, name = .name,
+             resultTime = .resultTime,
+             phenomenonTime = .phenomenonTime,
+             procedure = .procedure,
+             observableProperty = .observableProperty,
+             featureOfInterestType = .featureOfInterestType,
+             observationType = .observationType,
+             responseFormat = .responseFormat,
+             procedureDescriptionFormat = .procedureDescriptionFormat,
+             observedArea = .observedArea)
+
+  if(sos@verboseOutput)
+    cat("[parseSosObservationOffering] done: ", toString(.ob), "\n")
+
+  return(.ob)
+
+}
+
+
 parseGetObservationResponse <- function(obj, sos, verbose = FALSE) {
-  
+
   if(sos@verboseOutput) {
     cat("[parseGetObservationResponse] entering... \n")
     print(obj)
   }
-  
-  .offeringsXML <- .filterXmlChildren(
-    node = obj,
-    childrenName = "observationData")
-  .observations = sapply(.offeringsXML, parseObservation_2.0,
+
+  .observationsXML <- .filterXmlChildren(node = obj, xmlTagName = "observationData")
+  featureCache <<- list()
```

Comment:
We should avoid global assignments. I suggest to store the `featureCache` within the `sos` object.